src/includes/go.inc.js

```diff
@@ -360,6 +360,10 @@ function drupalgap_back() {
       });
     }
     else if (active_page_id == '_drupalgap_splash') { return; }
+    else if (drupalgap.back_path == ''){
```

Comment:
@mattshoaf I'm concerned here that if we don't call `_drupalgap_back()` we could experience some very odd behavior in a typical app when naturally navigating back to the front page. If you place a call to `_drupalgap_back()` on line 365, then return, do you still get your desired behavior. And even then, now that I look at it, I'm concerned the call to `drupalgap_goto()` on 364 would confuse the navigation system as well. Please elaborate a bit when you have a moment, thanks.